Java/core/src/main/java/com/amazon/randomcutforest/sampler/SimpleStreamSampler.java

```diff
@@ -74,7 +74,19 @@
      * The decay factor used for generating the weight of the point. For greater
      * values of lambda we become more biased in favor of recent points.
      */
-    private final double lambda;
+    private double lambda;
+    /**
+     * The last timestamp when lambda was changed
+     */
+    private long lastUpdateOflambda = 0;
+    /**
+     * most recent timestamp
+     */
+    private long largestSequenceIndexSeen = 0;
```

Comment:
This is hard-coding the assumption that sequence indexes will start at 0 and increase from there. That's consistent with what we do in the forest now, but we may need to think through this some more if we want to support something like using clock time as the index value.

Reply:
Clock time would be a natural next step. I was thinking of fixing the restriction in the HashSet for the sequenceIndices first before going to milliseconds/etc. Probably a subsequent PR.